### ISSUE TITLE
MedTagger should be easy to customize (POC)

### ISSUE THREAD

**Issue by jpowie01** [jpowie01/MedTagger#315]
## Expected Behavior

MedTagger should be configurable easily:
- from a CLI,
- or configuration files,
- or Admin Panel.

## Actual Behavior

MedTagger's Administrator needs to open the DB CLI or iPython to enter new Scan Category or a Tag or anything else. What's more, there is no easy way to add it without editing MedTagger's code!

Luckly, MedTagger can be set up easily with Docker containers, so part of this job is already done. This mechanism should be extended and containers should prepare a final set up based on given configuration.

## Additional comment

Let's prepare a POC and discuss it with others. Then, we should plan our work and create new Issues based on duscussion's results. Consider creating a milestone if there will be more than 3 Issues needed to be resolved.

**Comment by jpowie01** [comment]
As a result of this POC, I've come up with an example solution that may be very useful for administrators or data scientists that would like to use MedTagger. Final result consists of two main parts:
- CLI tool that helps in MedTagger's instance management,
- synchronization script that syncs YAML configuration with SQL DB entries.

Designed flow base on such a flow:
```bash
$> virtualenv -p python3.6 venv
$> . venv/bin/activate
(venv) $> pip install medtagger
(venv) $> medtagger init
(venv) $> medtagger config --generate
(venv) $> vim .medtagger.yml
(venv) $> medtagger config --validate
(venv) $> medtagger start
(venv) $> ...
(venv) $> medtagger stop
```

Overall solution base on YAML configuration that will be stored in `medtagger.yml` file. Here is an example of its content:

```yaml
tasks:
  - name: Kidneys segmentation
    key: KIDNEYS_SEGMENTATION
    tags:
        - name: Left Kidney
          key: LEFT_KIDNEY
          tools:
              - RECTANGLE
        - name: Right Kidney
          key: RIGHT_KIDNEY
          tools:
              - RECTANGLE

  - name: Find nodules
    key: FIND_NODULES
    tags:
        - name: Nodule
          key: NODULE
          tools:
              - RECTANGLE

  - name: Find narrowings in vains
    key: FIND_NARROWINGS
    tags:
        - name: Narrowing
          key: NARROWING
          tools:
              - RECTANGLE

datasets:
    - name: Kidneys
      key: KIDNEYS
      image_path: assets/icon/kidneys_category_icon.svg
      tasks:
        - KIDNEYS_SEGMENTATION

    - name: Lungs
      key: LUNGS
      image_path: assets/icon/lungs_category_icon.svg
      tasks:
        - FIND_NODULES

    - name: Heart
      key: HEART
      image_path: assets/icon/heart_category_icon.svg
      tasks:
        - FIND_NARROWINGS
```

Apart from above example, CLI tool will be able to handle multiple commands. These one needs to be done first:
- `medtagger init` - initialize and prepare your instance, pulls Docker Images based on preconfigured example Docker Compose, generates example `medtagger.yml` file,
- `medtagger config --generate` - generate simple example `medtagger.yml`,
- `medtagger config --validate` - validates `medtagger.yml` and its content,
- `medtagger start` - will start MedTagger instance based on Docker Compose file,
- `medtagger stop` - stops all MedTagger containers,
- `medtagger --help` - shows help for this CLI tool,
- `medtagger --version` - obvious.

In the future, we could add such commands:
- `medtagger generate` - will ask you some questions and generate final dataset with Labels,
- `medtagger update` - updates Docker Images,
- `medtagger upload` - uploads new Scans to the MedTagger instance from local disk,
- `medtagger deploy` - possible deployment using Ansible/Fabric files,
- `medtagger add user` - helps in creation of new accounts (possible first admin account),
- `medtagger backup --generate` - obvious,
- `medtagger backup --restore` - obvious.

As a result of this Issue, I've prepared a POC code that is available on `poc_for_simple_configuration_of_medtagger` branch:
https://github.com/jpowie01/MedTagger/tree/poc_for_simple_configuration_of_medtagger

But there is still a lot of work to do, which I've described in #336, #339, #340 and #341.
